pkg/kubelet/dockertools/docker.go

```diff
@@ -198,6 +199,109 @@ func (d *dockerContainerCommandRunner) RunInContainer(containerID string, cmd []
 	return buf.Bytes(), <-errChan
 }
 
+// ExecInContainer uses nsenter to run the command inside the container identified by containerID.
+//
+// TODO:
+//  - should we support `docker exec`?
+//  - should we support nsenter on the host? (current impl)
+//  - should we support nsenter in a container, running with elevated privs and --pid=host?
+func (d *dockerContainerCommandRunner) ExecInContainer(containerId string, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool) error {
+	container, err := d.client.InspectContainer(containerId)
+	if err != nil {
+		return err
+	}
+
+	if !container.State.Running {
+		return fmt.Errorf("container not running (%s)", container)
+	}
+
+	containerPid := container.State.Pid
+	args := append([]string{"-t", fmt.Sprintf("%d", containerPid), "-m", "-i", "-u", "-n", "-p", "--"}, cmd...)
```

Comment:
Does it makes sense to a.) find a way to make the process inherit the parent container's environment ([example](https://github.com/jpetazzo/nsenter/issues/18)) and/or b.) allow the client to inject their own environment variables?